### PR TITLE
20230413-more-wolfcrypt-c89

### PR DESCRIPTION
**Pull request by douzzer**
add `WOLF_C89` clauses to the `W64LIT()` definitions in `wolfssl/wolfcrypt/types.h`, and wrap several `long long` numeric literals with `W64LIT()` in `wolfcrypt/src/{aes.c,blake2b.c,siphash.c}`;

add `WOLF_C89` handling to `SP_ULONG_BITS` and `SP_ULLONG_BITS` setup in `wolfssl/wolfcrypt/sp_int.h`.

`wolfcrypt/src/fe_448.c`: fix `declaration-after-statement` in `fe448_mul_8()`.

fix "comma at end of enumerator list" warnings in wolfcrypt for C89 compatibility, mostly by just snipping out unneeded commas, but several using a C89-gated construct at the end of the enum definition:

```
#ifdef WOLF_C89
    x_c89_dummy_last_enum_val
#endif
```

### CODE REVIEW

**Review by dgarske — APPROVED** [review]

File: wolfcrypt/src/fe_448.c [hunk 1]
```diff
@@ -1836,6 +1836,8 @@ static WC_INLINE void fe448_mul_8(sword32* r, const sword32* a, const sword32* b
     sword64 t13  = (sword64)a[ 6] * b[ 7];
     sword64 t113 = (sword64)a[ 7] * b[ 6];
     sword64 t14  = (sword64)a[ 7] * b[ 7];
+    sword64 o;
```
Comment (dgarske) [inline wolfcrypt/src/fe_448.c:1839]: @SparkiDev , when you are back can you review the scripts repo to see if it needs this fix also?

File: wolfcrypt/src/sakke.c [hunk 2]
```diff
@@ -2575,16 +2584,35 @@ static int sakke_modexp_loop(SakkeKey* key, mp_int* b, mp_int* e, mp_proj* r,
 {
     int err;
 #ifdef WC_NO_CACHE_RESISTANT
-    mp_proj* c[2] = { r, key->tmp.p2 };
+    mp_proj* c[2]
```
Comment (dgarske) [inline wolfcrypt/src/sakke.c:2587]: Just make these the normal path too. No impact.

File: wolfssl/wolfcrypt/types.h [hunk 3]
```diff
@@ -112,16 +112,22 @@ decouple library dependencies with standard string, memory and so on.
...
-            || (defined(__cplusplus) && (__cplusplus >= 201103L))
+            || (defined(__cplusplus) && (__cplusplus >= 201103L)))
             #define HAVE_ANONYMOUS_INLINE_AGGREGATES 1
         #else
             #define HAVE_ANONYMOUS_INLINE_AGGREGATES 0
         #endif
     #endif
 
+    #if defined(WOLF_C89) || defined(WOLF_NO_TRAILING_ENUM_COMMAS)
+        #define WOLF_ENUM_DUMMY_LAST_ELEMENT(prefix) _wolf_ ## prefix ## _enum_dummy_last_element
```
Comment (dgarske) [inline wolfssl/wolfcrypt/types.h:126]: Please add note indicating when the dummy element is required for C89 and why it is only used in cases where there are complex macros above gating various conditions for the "last" element.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/sakke.c [hunk 4]
```diff
@@ -2228,11 +2228,20 @@ static int sakke_pairing(SakkeKey* key, ecc_point* p, ecc_point* q, mp_int* r,
     mp_proj* t2 = key->tmp.p3;
     mp_int* t3 = &key->tmp.m2;
     mp_int* prime = &key->params.prime;
-    mp_int* t[] = { &key->tmp.m1, t3 };
+    mp_int* t[]
+#ifndef WOLF_C89
+        = { &key->tmp.m1, t3 }
```
[the remark this answers is not in the document]
Comment (douzzer) [reply wolfcrypt/src/sakke.c:2233]: whoopsadoodle.  fixing.

File: wolfcrypt/src/sakke.c (hunk 2 above)
Comment (douzzer) [reply wolfcrypt/src/sakke.c:2587]: :heavy_check_mark:

File: wolfssl/wolfcrypt/types.h (hunk 3 above)
Comment (douzzer) [reply wolfssl/wolfcrypt/types.h:126]: :heavy_check_mark:

**Review by dgarske — APPROVED** [review]

**Pull request merged**